### PR TITLE
UGM Tree external group support.

### PR DESCRIPTION
**Pull request by jensens**

### CODE REVIEW

**Comment by rnixx** [comment]
Please keep bootstrap.sh, even if you're not using it.

**Comment by rnixx** [comment]
Is there a benefit with using a setup.cfg beside having another file?

File: CHANGES.rst [hunk 1]
```diff
@@ -1,10 +1,11 @@
 History
 =======
 
-1.0b12 (unreleased)
+1.0b13 (unreleased)
```
[the remark this answers is not in the document]
Comment (jensens) [reply CHANGES.rst:4]: ops, sure.

**Comment by jensens** [comment]
> Is there a benefit with using a setup.cfg beside having another file?

The fact you do not need to open and close files on setup time nor rely on Python code.
Aaaannd: Its declarative, don't we like declarative systems?

**Comment by jensens** [comment]
> Please keep bootstrap.sh, even if you're not using it.

It was just broken and copied stuff from a ghost ``.. `` place. Nobody cloning the repo knows what this should be. Kid of esoteric....

In this case we also need to fix it for python3 (``python -mvenv .``). 

And yes, I always use pyenv to avoid a bunch of problems and have my isolated python of a specific version.

So overall, I better copy a working ``bootstrap.sh`` from another project over here in order to make it work for everyone.

File: buildout.cfg [hunk 2]
```diff
@@ -1,3 +1,3 @@
 [buildout]
 extends = base.cfg
-develop = ..
```
Comment (jensens) [inline buildout.cfg:3]: @rnix btw.: this one here took me 30min to figure out... Why? Lets stick to the standards.

**Comment by rnixx** [comment]
It was not broken. And a pushd/popd is nothing esoteric. Anyway, there is no such thing as a bootstrap.sh standard. This is a thing some Plone people agreed to do in a specific form, and as you know I do almost no Plone nowadays. I'm fine if you copy your "standard" bootstrap.sh over here.

And setup.cfg is a thing I really not want to use unless there is a **real** reason to do so.

**Comment by jensens** [comment]
We should move to declarative setup.cfg for all packages step by step to get in the consistency. In the rare case code is needed those parts can still be handled by setup.py. But even then all other metadata resides then in setup.cfg. With wheels anyway nothing is executed on install time anymore.

This is official btw.: https://setuptools.readthedocs.io/en/latest/setuptools.html#configuring-setup-using-setup-cfg-files

IMO this is a dinosaur discussion. The reason is to move forward to modern Python and not stick to old habits. I never liked having metadata in Python code, its just wrong.

**Comment by jensens** [comment]
Regarding bootstrap.sh - no this is not a standard at all, its a non-pythonic thing from Plone. The one that was in here was just broken with undocumented preconditions. 

I tend to remove it everywhere. I think @tisto did it right by replacing all this with a Makefile in his projects. But this makes sense for complex projects. This one here is very pythonic and not Plone bound at all, so I would like to see even buildout kicked out here. Not that I plan to do this right now. nor do I have time.

**Comment by rnixx** [comment]
> IMO this is a dinosaur discussion.
yeah, and C is a dinosaur language. 
and python is a dinosaur as well.
and unix is a dinosaur OS
and linux has this dinosaur architecture too

your're right. all settled stuff is crap. thus let's be progressive. break out of a well maintained and unified form at one package out of tons just for the sake of it.

for me this is a non issue, i like sticking to one form or another, not to mix up stuff. also this is a non issue to me as it solves nothing.

seriously. if you want to spend days of work, got for it. but then do it everywhere at once and keep everything working, no regressions. if you just intend to get it into this one package, please stick to the dino. no evangelism please. i'd really appreciate.

i also really appreciate the rest of this PR by the way ;)

happy easter!

**Comment by rnixx** [comment]
merged relevant bits via 113d44f1851ec8b2467f555d5f8b8cd4eca7b97b

**Comment by rnixx** [comment]
I renamed this PR, since i merged back master to "external-groups" branch after picking related code, so this code not shows up any more in this PR.

@jensens referring the remaining package structure changes. As stated earlier, I'm not totally against modernization, I'm just against random changes (which not even work in this case by the way) in the stack. We'll find time and motivation at some sprint discussing and planning how to continue on this topic.

**Pull request closed without merging**